device/core/devdoc/device_twin_requirements.md

```diff
@@ -3,146 +3,75 @@
 
-**SRS_NODE_DEVICE_TWIN_18_020: [** `_sendTwinRequest` shall call `done` with an `err` value translated using `translateError`  **]** 
+**SRS_NODE_DEVICE_TWIN_16_006: [** For each desired property that is part of the `TwinProperties` object received, an event named after the path to this property shall be fired. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_021: [** Before calling `done`, `_sendTwinRequest` shall remove the handler for the `response` event **]** 
+### properties.reported.update(state: any, done: (err?: null) => void): void;
+`update` is a method which application developers use to send reported state to the service.
 
-**SRS_NODE_DEVICE_TWIN_18_022: [** If the response doesn't come within `Twin.timeout` milliseconds, `_sendTwinRequest` shall call `done` with `err`=`TimeoutError` **]**  
+**SRS_NODE_DEVICE_TWIN_16_007: [** The `update` method shall call the `updateReportedProperties` method of the `Transport` object and pass it the patch object and a callback accepting an error as argument. **]**
 
+**SRS_NODE_DEVICE_TWIN_16_008: [** If the callback passed to the transport is called with an error, the `callback` argument of the `update` method shall be called with that error. **]**
 
-### properties.reported.update method
-`update` is a method which application developers use to send reported state to the service.
+**SRS_NODE_DEVICE_TWIN_18_031: [** If the callback passed to the transport is called with no error, the  `properties.reported.update` shall merge the contents of the patch object into `properties.reported` **]**
 
-**SRS_NODE_DEVICE_TWIN_18_025: [** `properties.reported.update` shall use _sendTwinRequest to send the patch object to the service. **]** 
+**SRS_NODE_DEVICE_TWIN_18_032: [** When merging the patch, if any properties are set to `null`, `properties.reported.update` shall delete that property from `properties.reported`. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_026: [** When calling `_sendTwinRequest`, `properties.reported.update` shall pass `method`='PATCH', `resource`='/properties/reported/', `properties`={}, and `body`=the `body` parameter passed in to `reportState` as a string. **]**   
+**SRS_NODE_DEVICE_TWIN_16_009: [** Once the properties have been merged the `callback` argument of the `update` method shall be called with no argument. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_027: [** `properties.reported.update` shall call `done` with the results from `_sendTwinRequest` **]** 
+### on('properties.desired[.path])
 
-**SRS_NODE_DEVICE_TWIN_18_031: [** `properties.reported.update` shall merge the contents of the patch object into `properties.reported` **]**
+**SRS_NODE_DEVICE_TWIN_16_010: [** When a listener is added for an event name starting with `properties.desired` the `enableTwinDesiredPropertiesUpdates` method of the `Transport` object shall be called with a callback function accepting an optional error argument. **]**
```

Comment:
>shall be called with a callback function accepting an optional error argument [](start = 187, length = 77)

this requirement is confusing.  It's called "with a callback",  but what is that callback?  At first reading, I assumed it was related to the event handler callback, but I don't think that's true.